core/chapters/c06_lists.py

```diff
@@ -423,92 +423,319 @@ def check(self):
     class introducing_len_and_range(VerbatimStep):
         """
 There you go. `words[4]` and beyond don't exist, so trying that will give you an error.
-
-By the way, you can get the number of elements in a list (commonly called the *length*) using `len(words)`.
-That means that the last valid index of the list is `len(words) - 1`, so the last element is `words[len(words) - 1]`. Try these for yourself.
-
-So in general, the valid indices are:
-
-    [0, 1, 2, ..., len(words) - 2, len(words) - 1]
-
-There's a handy built in function called `range` which gives you these values. Try it by running this code:
+That first program is a bit repetitive. Let's improve it with a list and a loop!
 
 __program_indented__
         """
 
-        def program(self):
-            for i in range(10):
-                print(i)
-
-    class range_len(VerbatimStep):
-        """
-`range(n)` is similar to the list `[0, 1, 2, ..., n - 2, n - 1]`.
-This gives us an alternative way to loop over a list:
-
-__program_indented__
-        """
+        auto_translate_program = False
 
         predicted_output_choices = ["""\
+0
+1
+2
+3
 This
 is
 a
 list
-""", """\
-0
+""",  """\
 1
 2
 3
-""", """\
-0
+4
 This
-1
 is
-2
 a
-3
 list
 """, """\
 This
-0
 is
-1
 a
-2
 list
-3
-""", """\
 0
 1
 2
 3
+""", """\
+0
 This
+1
 is
+2
 a
+3
 list
 """, """\
+1
 This
+2
 is
+3
 a
+4
 list
+""",
+                                    ]
+
+        def program(self):
+            words = ['This', 'is', 'a', 'list']
+            indices = [0, 1, 2, 3]
+
+            for index in indices:
+                print(index)
+                print(words[index])
+
+    class using_range_instead_of_indices(VerbatimStep):
```

Comment:
```suggestion
    class range_len(VerbatimStep):
```

and rename the other `range_len`. I know the name doesn't make sense any more but it's fine.

Reply:
OK